src/main/kotlin/org/opensearch/replication/metadata/store/ReplicationMetadataStore.kt

```diff
@@ -200,6 +211,8 @@ class ReplicationMetadataStore constructor(val client: Client, val clusterServic
             throw ResourceNotFoundException("Metadata for $id doesn't exist")
         }
 
+        checkAndWaitForStoreHealth()
```

Comment:
nit : we might do with red index on `delete`, right ?

Reply:
Just thinking, waitForStoreHealth has dual benefits - waiting for the dependent plugins and shards to initialize. only case missing that will be missing here is deletion from red index. Should we handle at the caller end? and any way if the shard is not available this should result in error

Reply:
> if the shard is not available this should result in error

Agreed, so we are good here.